src/crypto.ts

```diff
@@ -140,129 +85,114 @@ function decrypt(signingPublicKey: string) {
       }
 
       if (verifiedContent) {
+        if (!this.publicSigningKey) {
+          throw new MissingPublicKeyError(
+            'Public signing key must be provided when instantiating the Crypto class in order to verify verified content'
+          )
+        }
         // Only care if it is the correct shape if verifiedContent exists, since
         // we need to append it to the end.
         // Decrypted message must be able to be authenticated by the public key.
-        const decryptedVerifiedContent = _decrypt(
+        const decryptedVerifiedContent = decryptContent(
           formSecretKey,
           verifiedContent
         )
         if (!decryptedVerifiedContent) {
-          // Returns null if verification for decrypt failed.
-          throw new Error('Verification failed for signature')
+          // Returns null if decrypting verified content failed.
+          throw new Error('Failed to decrypt verified content')
         }
-        const decryptedVerifiedObject = _verifySignedMessage(
+        const decryptedVerifiedObject = verifySignedMessage(
           decryptedVerifiedContent,
-          signingPublicKey
+          this.publicSigningKey
         )
 
         returnedObject.verified = decryptedVerifiedObject
       }
 
       return returnedObject
     } catch (err) {
+      // Should only throw if MissingPublicKeyError.
+      // This library should be able to be used to encrypt and decrypt content
+      // if the content does not contain verified fields.
+      if (err instanceof MissingPublicKeyError) {
+        throw err
+      }
       return null
```

Comment:
Seems odd that the caller of this function will not know the cause of the error?

Reply:
This was an early choice to follow how `tweetnacl` returns objects. If the encryption failed in anyway, then `null` is returned.